### PR TITLE
Add Scheduler field to the core/v2 Check and CheckConfig APIs

### PR DESCRIPTION
**Pull request by echlebek**
## What is this change?

This change adds the Scheduler field to the Check and CheckScheduler APIs.

## Why is this change necessary?

This field allows us to transmit information about how checks were scheduled to users. Checks can be scheduled with a variety of mechanisms in Sensu, including `memory`, `etcd`, and `postgres` (in the enterprise product).

## Does your change need a Changelog entry?

Yes

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Documentation will need updating if this change is accepted.

## How did you verify this change?

I verified that events now include this field in their check.

## Is this change a patch?

No

### CODE REVIEW

**Comment by echlebek** [comment]
@palourde can you please review?

File: backend/ringv2/ringv2.go [hunk 1]
```diff
@@ -463,25 +463,39 @@ func (r *Ring) startWatchers(ctx context.Context, ch chan Event, name string, va
...
-				if response.Canceled || !ok {
+				if !ok || response.Canceled {
 					// The watcher needs to be reinstated
-					r.startWatchers(ctx, ch, name, values, interval, cron)
-					return
+					if ctx.Err() == nil {
+						r.startWatchers(ctx, ch, name, values, interval, cron)
+						return
+					} else {
+						continue
```
Comment (palourde) [inline backend/ringv2/ringv2.go:476]: Should we use `notifyError` here and below on line 493?

File: backend/store/etcd/check_store.go [hunk 2]
```diff
@@ -26,6 +25,17 @@ func GetCheckConfigsPath(ctx context.Context, name string) string {
 	return checkKeyBuilder.WithContext(ctx).Build(name)
 }
 
+func schedulerFor(c *corev2.CheckConfig) string {
+	if c.Scheduler == "" {
+		if c.RoundRobin {
+			return "etcd"
```
Comment (palourde) [inline backend/store/etcd/check_store.go:31]: Should we use the constants here? e.g. `corev2.EtcdScheduler`

**Review by palourde — APPROVED** [review]
Changes look good, only two small suggestions

File: backend/ringv2/ringv2.go (hunk 1 above)
Comment (echlebek) [reply backend/ringv2/ringv2.go:476]: No, `ctx.Err() != nil` only means the context has been canceled. The next iteration will terminate the loop.

File: backend/store/etcd/check_store.go (hunk 2 above)
Comment (echlebek) [reply backend/store/etcd/check_store.go:31]: Yup, absolutely.

**Pull request merged**